### PR TITLE
sinkv2(ticdc): make table sink close cancellable

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/5928

### What is changed and how it works?
- Make table sink close cancellable
- rename to table_sink_impl.go
- add table ID for table sink

#### Tests 

 - Unit test

##### Will it cause performance regression or break compatibility?
No
##### Do you need to update user documentation, design documentation or monitoring documentation?
No
### Release note 

```release-note
None
```

### CODE REVIEW

**Review by hi-rustin — COMMENTED** [review]
Self check

File: cdc/sinkv2/tablesink/progress_tracker.go [hunk 1]
```diff
@@ -139,12 +150,27 @@ func (r *progressTracker) trackingCount() int {
...
+				zap.Int64("tableID", r.tableID),
+				zap.Int("trackingCount", r.trackingCount()),
+				zap.Any("lastMinResolvedTs", r.minTs()),
+			)
+		default:
+			if r.trackingCount() == 0 {
+				return nil
+			}
+			// Sleep for a while to prevent CPU spin.
+			time.Sleep(waitingInterval)
```
Comment (hicqu) [inline cdc/sinkv2/tablesink/progress_tracker.go:173]: How about `time.NewTicker(waitingInterval)`?

**Comment by hicqu** [comment]
Rest LGTM.

File: cdc/sinkv2/tablesink/progress_tracker.go [hunk 2]
```diff
@@ -52,8 +62,9 @@ type progressTracker struct {
 // newProgressTracker is used to create a new progress tracker.
 // The last min resolved ts is set to 0.
 // It means that the table sink has not started yet.
-func newProgressTracker() *progressTracker {
+func newProgressTracker(tableID model.TableID) *progressTracker {
```
Comment (nongfushanquan) [inline cdc/sinkv2/tablesink/progress_tracker.go:65]: why the tableID is needed here

Comment (hi-rustin) [reply cdc/sinkv2/tablesink/progress_tracker.go:65]: It's for log.

File: cdc/sinkv2/tablesink/progress_tracker.go (hunk 1 above)
Comment (hi-rustin) [reply cdc/sinkv2/tablesink/progress_tracker.go]: Good idea. Done!

**Review by hicqu — APPROVED** [review]

**Review by asddongmen — APPROVED** [review]

**Pull request merged**